aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -441,6 +444,150 @@ std::tuple<Tensor&,Tensor&> min_out(Tensor& min, Tensor& min_indices,
+  if (min && max) {
+    TORCH_CHECK(self.layout() == Layout::Strided,
+                "clamp only supports strided layout, got: ", self.layout());
+    auto iter = TensorIteratorConfig()
+                .set_check_mem_overlap(true)
+                .add_output(result)
+                .add_input(self)
+                .add_input(*min)
+                .add_input(*max)
+                .promote_inputs_to_common_dtype(true)
+                .cast_common_dtype_to_outputs(true)
+                .enforce_safe_casting_to_output(true)
+                .build();
+    clamp_stub(iter.device_type(), iter);
+  } else if (max) {
+    at::clamp_max_outf(self, *max, result);
+  } else if (min) {
+    at::clamp_min_outf(self, *min, result);
+  } else {
+    TORCH_CHECK(false, "At least one of 'min' or 'max' must not be None");
+  }
+  return result;
+}
+
+Tensor clamp(const Tensor& self, c10::optional<Scalar> min, c10::optional<Scalar> max) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_outf(self, min, max, result);
+}
+
+Tensor clamp(const Tensor& self, const c10::optional<Tensor>& min, const c10::optional<Tensor>& max) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_outf(self, min, max, result);
+}
+
+Tensor& clamp_(Tensor& self, c10::optional<Scalar> min, c10::optional<Scalar> max) {
+  return at::clamp_outf(self, min, max, self);
+}
+
+Tensor& clamp_(Tensor& self, const c10::optional<Tensor>& min, const c10::optional<Tensor>& max) {
+  return at::clamp_outf(self, min, max, self);
+}
+
+Tensor& clamp_max_out(const Tensor& self, Scalar max, Tensor& result) {
+  return at::clamp_max_outf(self, scalar_tensor_like(self, max), result);
+}
+
+Tensor& clamp_max_out(const Tensor& self, const Tensor& max, Tensor& result) {
+  TORCH_CHECK(self.layout() == Layout::Strided,
+              "clamp_max only supports strided layout, got: ", self.layout());
+  auto iter = TensorIterator::binary_op(result, self, max);
+  clamp_max_stub(iter.device_type(), iter);
+  return result;
+}
+
+Tensor clamp_max(const Tensor& self, Scalar max) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_max_outf(self, max, result);
+}
+
+Tensor clamp_max(const Tensor& self, const Tensor& max) {
+  Tensor result = at::empty({0}, self.options());
+  return at::clamp_max_outf(self, max, result);
+}
+
+Tensor& clamp_max_(Tensor& self, Scalar max) {
+  return at::clamp_max_outf(self, max, self);
+}
+
+Tensor& clamp_max_(Tensor& self, const Tensor& max) {
+  return at::clamp_max_outf(self, max, self);
+}
+
+Tensor& clamp_min_out(const Tensor& self, Scalar min, Tensor& result) {
+  return at::clamp_min_outf(self, scalar_tensor_like(self, min), result);
+}
+
+Tensor& clamp_min_out(const Tensor& self, const Tensor& min, Tensor& result) {
+  TORCH_CHECK(self.layout() == Layout::Strided,
+              "clamp_min only supports strided layout, got: ", self.layout());
```

Comment:
"torch.clamp() only supports..."

since the name "clamp_min" is internal only (so it's only appropriate to use with TORCH_INTERNAL_ASSERT, for example